src/components/pages/ShopDetail.tsx

```diff
@@ -0,0 +1,268 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+// import { Link } from "react-router-dom";
+import { GoogleMap, LoadScript, Marker, InfoWindow } from "@react-google-maps/api";
+import axios from "axios";
+
+import { styled } from '@mui/material/styles';
+import Container from "@mui/material/Container";
+import Grid from "@mui/material/Grid";
+import Card from "@mui/material/Card";
+import CardActions from "@mui/material/CardActions";
+import CardContent from "@mui/material/CardContent";
+import CardMedia from "@mui/material/CardMedia";
+import CardHeader from '@mui/material/CardHeader';
+import Collapse from '@mui/material/Collapse';
+import Button from '@mui/material/Button';
+import IconButton, { IconButtonProps } from '@mui/material/IconButton';
+import Typography from "@mui/material/Typography";
+import Rating from "@mui/material/Rating";
+import FavoriteIcon from '@mui/icons-material/Favorite';
+import ChatBubbleIcon from '@mui/icons-material/ChatBubble';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+
+// <Map基本情報>
+const containerStyle = {
+  height: "60vh",
+  width: "100%"
+};
+// infoWindow style
+const divStyle = {
+  background: "white",
+  fontSize: 7.5
+};
+// </Map基本情報>
+
+// <折り畳み機能>
+interface ExpandMoreProps extends IconButtonProps {
+  expand: boolean;
+}
+
+const ExpandMore = styled((props: ExpandMoreProps) => {
+  const { expand, ...other } = props;
+  return <IconButton {...other} />;
+})(({ theme, expand }) => ({
+  transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
+  marginLeft: 'auto',
+  transition: theme.transitions.create('transform', {
+    duration: theme.transitions.duration.shortest,
+  }),
+}));
+//</折り畳み機能>
+
+function ShopDetail() {
+  // params id を受け取る
+  const { id } = useParams();
+  // 折り畳み機能 on/off
+  const [expanded, setExpanded] = React.useState(false);
+  const handleExpandClick = () => {
+    setExpanded(!expanded);
+  };
+  // shopデータの型
+  const [shop, setShop] = useState<Shop>();
```

Comment:
初期値はundefinedっぽそうなのでこんな感じですかね？？
```suggestion
  const [shop, setShop] = useState<Shop | undefined>(undefined);
```